dune_client/api/extensions.py

```diff
@@ -0,0 +1,189 @@
+"""
+Extended functionality for the ExecutionAPI
+"""
+
+import time
+from typing import Union, Optional, Any
+
+from deprecated import deprecated
+
+from dune_client.api.execution import ExecutionAPI
+from dune_client.models import (
+    ResultsResponse,
+    DuneError,
+    ExecutionState,
+    QueryFailed,
+    ExecutionResultCSV,
+)
+from dune_client.query import QueryBase
+
+
+class ExtendedAPI(ExecutionAPI):
```

Comment:
We could go with this, but in the very near future we will be adding also `run_sql` which is an extended feature but also uses the `queryAPI`. This would mean we either need to rename it again soon, or then extend to have 

`ExtendedQueryAPI`, ExtendedExectionAPI` and `ExtendedMixed` which is both extended. (since run_sql uses both). Then there is no `ExtendedQueryAPI` because it has no functions... Simplest design here is to leave it as `ExtendedAPI` which extends both the query and exection API. At some point you may want a clearer separation between `DuneClient` and `PremiumDuneClient`, but even then... the naming conventions are not completely obvious.

I would leave it as is... unless you insist.